db/counting_index.go

```diff
@@ -201,6 +204,7 @@ func (c *countingIndex) UseBatch(b batch.KVStoreBatch) error {
 		return ErrInvalid
 	}
 	c.batch = b
+	c.batch.AddFillPercent(c.bucket, 1.0)
```

Comment:
this changes previous logic

Reply:
before the batch is committed, AddTotalSize() will be called to update the total size, that is a more proper place to set the FillPercent. so change AddTotalSize() to Finalize() and move it there